great_expectations/checkpoint/actions.py

```diff
@@ -300,6 +300,73 @@ def _root_validate_slack_params(cls, values: dict) -> dict:
 
         return values
 
+    @override
+    def v1_run(
+        self, checkpoint_result: CheckpointResult, action_context: ActionContext | None = None
+    ) -> dict:
+        success = checkpoint_result.success or False
+        result = {"slack_notification_result": "none required"}
+        if not self._is_enabled(success=success):
+            return result
+
+        checkpoint_text_blocks: list[dict] = []
+        for (
+            validation_result_suite_identifier,
+            validation_result_suite,
+        ) in checkpoint_result.run_results.items():
+            validation_text_blocks = self._render_validation_result(
+                result_identifier=validation_result_suite_identifier,
+                result=validation_result_suite,
+                action_context=action_context,
+            )
+            checkpoint_text_blocks.extend(validation_text_blocks)
+
+        query = self.renderer.build_query(
+            checkpoint_result=checkpoint_result, text_blocks=checkpoint_text_blocks
+        )
+        blocks = query.get("blocks")
```

Comment:
the flow here with `query` and `blocks` doesn't feel intuitive to me. I also left comments on this below in `build_query`. The way this is structured, it's not clear that `query.get_blocks is checkpoint_text_blocks`, but it looks like that is the case.